ziggurat-core-geoip/src/providers/ip2loc.rs

```diff
@@ -13,21 +13,36 @@ use crate::{
 pub struct Ip2LocationService {
     /// Database file path.
     pub database_file: String,
+
+    /// For LITE versions IPv4 and IPv6 databases are separated. If you want to use IPv6 database,
+    /// you need to specify the path to the IPv6 database file.
+    /// NOTE: commercial version have both IPv4 and IPv6 data in one database file. When using
+    /// commercial version, you don't need to specify this field.
+    pub database_file_ipv6: Option<String>,
 }
 
 impl Ip2LocationService {
-    pub fn new(database_file: &str) -> Self {
+    pub fn new(database_file: &str, database_file_ipv6: Option<String>) -> Self {
         Self {
             database_file: database_file.to_owned(),
+            database_file_ipv6,
         }
     }
 }
 
 #[async_trait]
 impl GeoIPService for Ip2LocationService {
     async fn lookup(&self, ip: IpAddr) -> Result<GeoIPInfo, String> {
+        let db_file = ip.is_ipv4().then_some(&self.database_file).or({
+            match &self.database_file_ipv6 {
+                Some(file) => Some(file),
+                None => Some(&self.database_file), // if no IPv6 database file is specified, use the general one provided
+            }
+        });
+
+        // We can safely unwrap here as if ipv6 database would be empty we'll use general one
         let mut db =
-            DB::from_file(&self.database_file).map_err(|_| "database file can't be loaded")?;
+            DB::from_file(db_file.unwrap()).map_err(|_| "database file can't be loaded")?;
```

Comment:
Suggestions:

**a)** We could reach this part with `db_file` already being a `String`, not an `Option<String>`, so the extra `unwrap` is not necessary - by parsing ip above with match and then parsing directly:
```rust
pub enum IpAddr {
    V4(Ipv4Addr),
    V6(Ipv6Addr),
}
``` 

**alternative b)** keep it as it is and use `.map_err("no db file found")?` here instead of `unwrap()` (which crashes the program)

Reply:
Done - it should now be a bit shorter.